Apps/W1/EDocumentsConnector/app/src/Continia/API Requests/APIRequests.Codeunit.al

```diff
@@ -0,0 +1,1072 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace Microsoft.EServices.EDocumentConnector.Continia;
+using Microsoft.eServices.EDocument;
+using System.Utilities;
+using System.Text;
+using Microsoft.Foundation.Address;
+using Microsoft.Foundation.Company;
+
+codeunit 6393 "API Requests"
+{
+    Access = Internal;
+
+    Permissions = tabledata "Network Profile" = rimd,
+                  tabledata "Network Identifier" = rimd,
+                  tabledata "Participation" = rimd,
+                  tabledata "Activated Net. Prof." = rimd;
+
+    #region Get Network Profiles from Continia Delivery Network API
+
+    [InherentPermissions(PermissionObjectType::TableData, Database::"Network Profile", 'rimd', InherentPermissionsScope::Both)]
+    internal procedure GetNetworkProfiles(Network: Enum "Network")
+    var
+        APIUrlMgt: Codeunit "API URL Mgt.";
+        IsLastPage: Boolean;
+        HttpResponseMessage: HttpResponseMessage;
+        CurrPage: Integer;
+        PageSize: Integer;
+    begin
+        CurrPage := 0;
+        PageSize := 100;
+        IsLastPage := false;
+        repeat
+            CurrPage += 1;
+            if ExecuteRequest('GET', APIUrlMgt.NetworkProfilesURL(Network, CurrPage, PageSize), HttpResponseMessage) then
+                IsLastPage := HandleNetworkProfileResponse(HttpResponseMessage, Network, PageSize);
+        until IsLastPage;
+    end;
+
+    [InherentPermissions(PermissionObjectType::TableData, Database::"Network Profile", 'rimd', InherentPermissionsScope::Both)]
```

Comment:
Delete permissions are not needed for Network Profile.

Reply:
Should be fixed throughout

Reply:
After further considerations, we chose to remove the Inherent permission as we use Indirect permissions on the objects already.